Reduce latency establishing desktop connections

rosstimothy opened the pull request:
Eliminates the Proxy communicating with itself via SSH to generate user TLS certificates when connecting to desktops. The web api already has an establish auth client to the root and leaf cluster and is capable of issuing certificates directly with them.

zmb3 reviewed (approved):
Change looks good.

It might be nice to move the tracing stuff into a separate PR so we don't lose it if we end up having to revert, but up to you.

espadolini commented on lib/web/desktop.go:
```diff
@@ -256,72 +223,180 @@ const (
+	defer func() {
 		if err != nil {
-			return nil, trace.Wrap(err)
+			span.RecordError(err)
```
`RecordError` is a noop when the error is nil, right?

espadolini reviewed (approved):
Should we be slightly worried that we're duplicating the MFA ceremony code, or is it simple enough and/or different enough for the websocket UI that it's ok to have it here?

rosstimothy committed:
Reduce latency establishing desktop connections

Eliminates the Proxy communicating with itself via SSH to generate
user TLS certificates when connecting to desktops. The web api
already has an establish auth client to the root and leaf cluster
and is capable of issuing certificates directly with them.
